SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift

```diff
@@ -38,24 +37,30 @@ final class NetworkRequestor: Dependency {
     
     // MARK: - Public methods
     
-    public func fetchDecodable<T: Decodable>(_ url: String, completion: @escaping (Result<T, AFError>) -> Swift.Void) {
+    public func fetchDecodable<T: Decodable>(_ url: String, completion: @escaping (Result<T, Swift.Error>) -> Swift.Void) {
         let request = self.session.request(url, headers: self.headers)
         request.responseDecodable(of: T.self, queue: self.concurrentQueue) { (response) in
-            DispatchQueue.mainAsync { completion(response.result) }
+            DispatchQueue.mainAsync { completion(response.result.mapError({$0.swiftError})) }
         }
     }
     
-    public func fetchData(_ url: String, completion: @escaping (Result<Data, AFError>) -> Swift.Void) {
+    public func fetchData(_ url: String, completion: @escaping (Result<Data, Swift.Error>) -> Swift.Void) {
         let request = self.session.request(url, headers: self.headers)
         request.responseData(queue: self.concurrentQueue) { (response) in
-            DispatchQueue.mainAsync { completion(response.result) }
+            completion(response.result.mapError({$0.swiftError}))
         }
     }
     
-    public func fetchJSON(_ url: String, completion: @escaping (Result<Any, AFError>) -> Swift.Void) {
+    public func fetchJSON(_ url: String, completion: @escaping (Result<Any, Swift.Error>) -> Swift.Void) {
         let request = self.session.request(url, headers: self.headers)
         request.responseJSON(queue: self.concurrentQueue) { (response) in
-            DispatchQueue.mainAsync { completion(response.result) }
+            completion(response.result.mapError({$0.swiftError}))
```

Comment:
Can you we the `mapError()` part into a reusable extension?

Reply:
If i understant correctly `mapError()` should be an extension function to the Alamofire result?

Reply:
`mapError` should be on the default `Result` object Apple built us?

Reply:
But there are some instances where AF added their own `mapError`?

Reply:
> `mapError` should be on the default `Result` object Apple built us?

exactly it is, and that's the one we're using here

Reply:
Right, but I am looking for something like `response.result.swiftError` so we dont have `mapError({$0.swiftError})` in N places.

Reply:
> Right, but I am looking for something like `response.result.swiftError` so we dont have `mapError({$0.swiftError})` in N places. Since in reality this is all doing the same thing `AFError` -> `Swift.Error`

Now I got it, great will do it RN and commit.

Reply:
Just a thought. Not really anything big. We can move on to later.

Reply:
If it's not really a big thing then let's move on and ill work on the extension on a separate branch.

Reply:
I managed to make it look like this `completion(response.result.swiftError)`

Reply:
sounds great! adding here?

Reply:
too late :)  hope you have time to make another PR 🙇

Reply:
nope, another branch.